### PR TITLE
Pull request for branch 'addremove' that contains complete To Do app functionality

### PR DESCRIPTION
**Pull request by jkanyi-web**
- Fixed Linter errors
- Made file src/index.js that contains functionality for To Do App
- Made file src/main.css that contains styling for To Do App
- Asjusted contents of src/index.html to accomodate refactored To Do App

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,36 +1,108 @@
+/* eslint-disable no-unused-vars */
```
Comment (emmiiorji) [inline src/index.js]: Kindly ensure that you do not disable linter checks inline. They represent coding standards that you share with your team and most of the time, help to enforce best coding practices.

File: src/styles/main.css [hunk 2]
```diff
@@ -33,32 +26,28 @@ header {
...
 }
 
+.task-input img {
+  width: 30px;
+  height: 30px;
+  position: absolute;
+  top: 24%;
+  left: 10%;
+}
+
```
Comment (emmiiorji) [inline src/styles/main.css:36]: You have a beautiful design for your todo list. However, launching the project using the dev script (`npm run dev`) does not seem to compile the CSS properly as it does not have styling. This makes it impossible to use the beautiful features you have. Kindly fix this.

I could see the styling on by running the `src/index.html` file in live server.

Project on live server:
image

Launching with script:
image

**Review by emmiiorji — CHANGES_REQUESTED** [review]
Hi @jkanyi-web, 
# Status: Changes Required ♻️

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<div align='center'>

</div>

## Good Highlights✔️

- Linter checks are passing
- Good commit messages
- Good design
- Repo is professional


## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @jkanyi-web 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - All changes requested have been implemented  ✅ 
 - Good todo list app design 👏🏿  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 3]
```diff
@@ -1,36 +1,147 @@
...
+  const taskName = selectedTask.parentElement.lastElementChild;
+  if (selectedTask.checked) {
+    taskName.classList.add('checked');
+    todos[selectedTask.id].status = 'completed';
+  } else {
+    taskName.classList.remove('checked');
+    todos[selectedTask.id].status = 'pending';
+  }
+  localStorage.setItem('todo-list', JSON.stringify(todos));
+};
```
Comment (codepantha) [inline src/index.js:123]: - [ ] According to requirements, please create a new `js` file imported as a module that contains methods related to the status updates `(completed: true / false)`
- [ ] The functionality to mark a to-do item as completed or not is not working as expected. Currently nothing happens when the checkbox is clicked. Please implement this requirement 🙏

**Comment by jkanyi-web** [comment]
Hi Emmanuel. Thanks for the feedback and positive response.
My project has not been reviewed for a while now. Can you please also
review the remaining part of the project so I can complete my week?

File: src/index.js [hunk 4]
```diff
@@ -1,36 +1,147 @@
...
-          <label for="${task.index}">
-            <input type="checkbox" name="task" id="${task.index}">
-            <p>${task.description}</p>
-          </label>
-            <i class="fa-solid fa-ellipsis-vertical"></i> 
-              `;
-  taskBox.append(list);
+      const taskInfo = { name: userTask, status: 'pending' };
+      // adding a new task to todos
+      todos.push(taskInfo);
```
Comment (codepantha) [inline src/index.js:137]: - [ ] Here, where you create the to-do item, please ensure to be consistent with the keys. In this case, the `index` key is missing. The index is is important especially when you want to delete or update an item. 
- [x] Also, kindly change the `status` key to `completed` and ensure that it is set to `false` for any new to-do item that is created 👍

File: src/index.js [hunk 5]
```diff
@@ -1,36 +1,147 @@
...
+    };
+  }
+}
+
+clearAll.addEventListener('click', () => {
+  // Removing all tasks from array/todos
+  todos.splice(0, todos.length);
+  localStorage.setItem('todo-list', JSON.stringify(todos));
+  showTodo('all');
+});
```
Comment (codepantha) [inline src/index.js:109]: - [ ] This functionality is meant to delete all to-do items marked as `completed: true` using the `Array.filter()` method. At the moment, you are removing the entire to-do list which is not the expected behaviour.

**Review by codepantha — COMMENTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @jkanyi-web 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**